crates/vim/src/motion.rs

```diff
@@ -707,22 +708,22 @@ impl Motion {
         (new_point != point || infallible).then_some((new_point, goal))
     }
 
-    // Expands a selection using self motion for an operator
-    pub fn expand_selection(
+    pub fn range(
         &self,
         map: &DisplaySnapshot,
-        selection: &mut Selection<DisplayPoint>,
+        cur_selection: &mut Selection<DisplayPoint>,
```

Comment:
Does this still have to be mutable if you return a new range? Interface seems to be a bit confusing: we pass in a selection that's possibly mutated _and_ we get a new selection back?

Reply:
You're right, I want to try not to change the original logic, ignore some parameters, I think we may need a method, to get the range that the current motion corresponds to will be selected, at present, if we want to get the range, we can only call the expand method, although this can get the corresponding range, but it will also make the current vim state become selected, it would be nice to have a method that can be called at will to get the range, and there are no side effects

Reply:
> Does this still have to be mutable if you return a new range? Interface seems to be a bit confusing: we pass in a selection that's possibly mutated _and_ we get a new selection back?

You're right, I want to try not to change the original logic, ignore some parameters, I think we may need a method, to get the range that the current motion corresponds to will be selected, at present, if we want to get the range, we can only call the expand method, although this can get the corresponding range, but it will also make the current vim state become selected, it would be nice to have a method that can be called at will to get the range, and there are no side effects, I'm going to be making some adjustments